langserver/completion.go

```diff
@@ -59,16 +63,21 @@ func (h *LangHandler) handleTextDocumentCompletion(ctx context.Context, conn jso
 		case "var":
 			kind = lsp.CIKVariable
 		}
+
+		itf, newText := h.getNewText(kind, it.Name, it.Type)
 		citems[i] = lsp.CompletionItem{
-			Label:  it.Name,
-			Kind:   kind,
-			Detail: it.Type,
+			Label:            it.Name,
+			Kind:             kind,
+			Detail:           it.Type,
+			InsertTextFormat: itf,
+			// InsertText is deprecated in favour of TextEdit, but added here for legacy client support
+			InsertText: newText,
```

Comment:
Would you want to keep `InsertText` around for older clients?

Reply:
My vote is to keep this around for the time being.  The Atom client does not yet read snippet content from `CompletionItem.textEdit`.  I am not sure about other clients.

I've opened a [PR](https://github.com/atom/atom-languageclient/pull/165) to fix it.